### ISSUE TITLE
System-tests: missing screenshots in marathon test-suite

### ISSUE THREAD

**Issue by tkratky** [EnMasseProject/enmasse#1532]
In marathon test-suite when some web console related test fail then no screenshots are stored.
last time it happened in `io.enmasse.systemtest.marathon.StandardMarathonTest.testCreateDeleteAddressesViaAgentLong`

```
08:12:20,808 [main] INFO io.enmasse.systemtest.selenium.SeleniumProvider - Saving browser console log...
08:12:20,808 [main] WARN io.enmasse.systemtest.selenium.SeleniumProvider - Cannot save browser log: null
08:12:20,808 [main] INFO io.enmasse.systemtest.selenium.SeleniumProvider - Taking screenshot
08:12:20,808 [main] WARN io.enmasse.systemtest.selenium.SeleniumProvider - Cannot take screenshot: null
08:12:20,808 [main] WARN io.enmasse.systemtest.selenium.SeleniumProvider - Cannot save screenshots: null
```

**Comment by RH-sdavey** [comment]
from Marathon test : io.enmasse.systemtest.marathon.BrokeredMarathonTest.testTopicPubSubLong

46:39,865 [main] INFO io.enmasse.systemtest.amqp.AmqpClientFactory - External endpoint: messaging-test-topic-pubsub-brokered.127.0.0.1.nip.io:443, internal: messaging-test-topic-pubsub-brokered.127.0.0.1.nip.io:443
...successful output omitted....
01:47:49,820 [main] INFO io.enmasse.systemtest.marathon.MarathonTestBase - Client is closed.
01:47:50,843 [main] INFO io.enmasse.systemtest.amqp.AmqpClientFactory - External endpoint: messaging-test-topic-pubsub-brokered.127.0.0.1.nip.io:443, internal: messaging-test-topic-pubsub-brokered.127.0.0.1.nip.io:443
01:47:50,870 [vert.x-eventloop-thread-0] INFO io.enmasse.systemtest.amqp.ClientHandlerBase - Connection to messaging-test-topic-pubsub-brokered.127.0.0.1.nip.io:443 failed: null
01:49:50,845 [main] INFO io.enmasse.systemtest.marathon.MarathonTestBase - Client is closed.
01:49:50,855 [main] INFO io.enmasse.systemtest.selenium.SeleniumProvider - Saving browser console log...
01:49:50,856 [main] WARN io.enmasse.systemtest.selenium.SeleniumProvider - Cannot save browser log: null
01:49:50,856 [main] INFO io.enmasse.systemtest.selenium.SeleniumProvider - Taking screenshot
01:49:50,857 [main] WARN io.enmasse.systemtest.selenium.SeleniumProvider - Cannot take screenshot: null
01:49:50,857 [main] WARN io.enmasse.systemtest.selenium.SeleniumProvider - Cannot save screenshots: null
01:49:50,857 [main] INFO io.enmasse.systemtest.selenium.SeleniumProvider - Tear down selenium web drivers

Seems to me that the screenshots are not able to be taken because the connection to enmasse deployment failed. Will investigate why further.

**Comment by kornys** [comment]
@RH-sdavey I disagree. Im guessing that problem is in wrong log path, or list is empty or something different.

**Comment by RH-sdavey** [comment]
@tkratky @kornys 
Each test is failing exactly 30 minutes after the 30 minute test loop is started. I believe that if the test is in middle of one loop when the 30 minute timer is finished, the deployment is torn down, and not able to connect, therefore no logs are able to be saved, and no screenshots are possible. 
I think its not really a problem of wrong path, as before the exact 30 minute deadline, all screenshots and logs are saving with no problem, exactly after 30 minutes, there starts to be problems with connections and therefore saving screenshots etc.

**Comment by RH-sdavey** [comment]
for example : 
io.enmasse.systemtest.marathon.BrokeredMarathonTest.testTopicPubSubLong-STARTED
01:18:00,548 [main] INFO io.enmasse.systemtest.marathon.MarathonTestBase - Starting test running for 30 minutes at Thu Aug 23 01:18:00 CEST 2018

...all screenshots and logs are saving correctly...
...30 minutes later....

01:47:50,870 [vert.x-eventloop-thread-0] INFO io.enmasse.systemtest.amqp.ClientHandlerBase - Connection to messaging-test-topic-pubsub-brokered.127.0.0.1.nip.io:443 failed: null
01:49:50,845 [main] INFO io.enmasse.systemtest.marathon.MarathonTestBase - Client is closed.
01:49:50,855 [main] INFO io.enmasse.systemtest.selenium.SeleniumProvider - Saving browser console log...
01:49:50,856 [main] WARN io.enmasse.systemtest.selenium.SeleniumProvider - Cannot save browser log: null
01:49:50,856 [main] INFO io.enmasse.systemtest.selenium.SeleniumProvider - Taking screenshot
01:49:50,857 [main] WARN io.enmasse.systemtest.selenium.SeleniumProvider - Cannot take screenshot: null

**Comment by kornys** [comment]
@RH-sdavey our selenium implementation stores screenshots only if test failed, so during test it does not store screenshots I guess

**Comment by kornys** [comment]
BTW: When selenium stores and saves screenshots, it logged this operation on output.

**Comment by kornys** [comment]
@RH-sdavey and tests are not stopped after 30 minutes, but if tests run ends after 30 minutes from start, next run won't be run.

**Comment by RH-sdavey** [comment]
@kornys  Im sure I saw some failing Marathon test output that showed many screenshots were being taken during the actual test, and then couldnt save a screenshot after the 30 minute deadline... but I can find it now, will search more for it. 

and sure, I understand the logic of the 30 minute test loop, got that, but what I think is happening is that if there is a test run happening when the 30 minute deadline happens... somehow the active deployment is torn down or not able to connect any further ( even while the test is still running ), therefore not being able to take the screenshot. Certainly thats what I saw in the test mentioned above, will search for it again. 

Maybe Im going in the completely wrong direction, but it seems that is what is happening to me.
